### PR TITLE
DEX Part 3

### PR DESCRIPTION
**Pull request by jmenon**

### CODE REVIEW

File: src/ngraph/runtime/cpu/cpu_builder.hpp [hunk 1]
```diff
@@ -29,6 +29,134 @@
...
+    else if (R == 12)                                                                              \
+        KV = K<ET, 12>;                                                                            \
+    else if (R == 13)                                                                              \
+        KV = K<ET, 13>;                                                                            \
+    else if (R == 14)                                                                              \
+        KV = K<ET, 14>;                                                                            \
+    else if (R == 15)                                                                              \
+        KV = K<ET, 15>;                                                                            \
+    else if (R == 16)                                                                              \
+        KV = K<ET, 16>;
```
Comment (jbobba) [inline src/ngraph/runtime/cpu/cpu_builder.hpp]: Why not KV = K<ET, R>?

File: src/ngraph/runtime/cpu/cpu_builder.hpp [hunk 2]
```diff
@@ -29,6 +29,134 @@
...
+    }                                                                                              \
+    else if (ET == element::u32)                                                                   \
+    {                                                                                              \
+        SELECT_RANK(KV, uint32_t, R, K);                                                           \
+    }                                                                                              \
+    else if (ET == element::u64)                                                                   \
+    {                                                                                              \
+        SELECT_RANK(KV, uint64_t, R, K);                                                           \
+    }
+
```
Comment (jbobba) [inline src/ngraph/runtime/cpu/cpu_builder.hpp]: i'm assuming that this will be used in future PRs?

Comment (jmenon) [reply src/ngraph/runtime/cpu/cpu_builder.hpp]: Yes, just wanted to get the header change in earlier with the refactor

File: src/ngraph/runtime/cpu/cpu_builder.hpp (hunk 1 above)
Comment (jmenon) [reply src/ngraph/runtime/cpu/cpu_builder.hpp]: To allow for cases where R is not a constexpr, and the compiler should eliminate these when it is

**Pull request merged**